Development into Main

Charlie-robin opened the pull request:
Hey,

This is the PR for the migration of [algo-dojo-js](https://github.com/corndel-swe/algo-dojo-js)  to Java.

The main things to check are:
- That the project builds correctly, so you're able to fetch dependencies and run the benchmarks and unit tests.
  - See README
- That the README and instructions are clear and make sense.

The solutions can be found on the `solutions` branch.

Charlie-robin committed:
Added JMH & Junit, working with basic stubs

Charlie-robin committed:
Port Sorting & Benchmark config

Charlie-robin committed:
Started Trees :S

Charlie-robin committed:
Ported CountCompleted :)

Charlie-robin committed:
Change Task record to class

Charlie-robin committed:
Rename & Added MarkAllDone

Charlie-robin committed:
Added TaskCounter

Charlie-robin committed:
Removed TaskMapper.taskToString

Charlie-robin committed:
Moved MaxEffortFinder over

Charlie-robin committed:
Moved EffortTotaller over

Charlie-robin committed:
Refactor, Added Static & Better use of JMH for trees

Charlie-robin committed:
MaxWaterContainer moved

Charlie-robin committed:
PairSum added

Charlie-robin committed:
ArrayFilter added

Charlie-robin committed:
ArrayReverser added

Charlie-robin committed:
ZeroShifter added

Charlie-robin committed:
Naive removed

Charlie-robin committed:
Added images

Charlie-robin committed:
Fix run benchmark via CLI

Charlie-robin committed:
Added mvnw & started updating README.md

Charlie-robin committed:
Deleted Benchmark & unit test tester classes

Charlie-robin committed:
Updated Kata Instructions, README.md & Renamed method names

Charlie-robin committed:
Removed solutions

nimonian commented on README.md:
````diff
@@ -1 +1,157 @@
+   ```bash
+     ./mvnw clean verify
+   
+     java -jar target/benchmarks.jar SortingBenchmark
````
I needed to run `./mvnw package` before this would work

nimonian reviewed (commented):
- Everything worked except the not I made - I needed to run `./mvnw package` to get the jar before I could run the benchmarks

- It might be worth adding a note to say that when they do `./mvnw clean install` they will see lots of failing tests - that's cool cos the methods under test aren't implemented yet (that's your job!)

Charlie-robin committed:
Update README.md

Charlie-robin commented:
@nimonian Yes, good spot on both! 

I was running the commands with the solutions, so I missed the failing tests 😅. 

I've added a sentence explaining this and updated the Maven wrapper command to use `package` instead of `clean verify`. I think as I was using IntelliJ I had already built the jar so for some reason thought `verify` did that. 

I'll go ahead and merge, as both amendments have been made. 

Thanks for the review and the test run 😁
